### ISSUE TITLE
Unusual battery drain, wrong configuration?

### ISSUE THREAD

**Issue by TheCrypt0** [daliansky/XiaoMi-Pro-Hackintosh#224]
Hi,
First of all, **big up for your work**! I successfully managed to install macOS Mojave on my Mi Notebook Pro i7 16GB, everything is working fine (the trackpad is **awesome**) except for the battery life.

I usually get around 3/3.5 hours out of the machine while doing basic stuff, on Windows this was 6/7 hours. I'm pretty sure there's something wrong in my configuration but I really cannot find out what it is.

In other threads (https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/116 and https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/97) people claim that their idle power is around 5/10 watts. I just checked mine and it's 18/19 watts idle (with chrome opened and backlight set to 50%, the overall cpu usage is less than 5%).

I'm on the latest BIOS version with the downvoltage patch (https://github.com/Cr0wTom/Mi-Notebook-Pro-Mods) applied, voltageshift is set to -150 for CPU, GPU and cache (same as I'm using on Xtu).

![voltageshift_mon](https://user-images.githubusercontent.com/45895254/61723876-21b41000-ad6d-11e9-9c75-f61db5d5b3d0.png)

BIOS info:

![bios_screenshot](https://user-images.githubusercontent.com/45895254/61724111-938c5980-ad6d-11e9-8bae-f43775deb69a.jpg)

I haven't tried the CFG unlock patch (https://github.com/Cr0wTom/Mi-Notebook-Pro-Mods/blob/master/Patches/CFG_unlock.cmd) yet. Is it known to make any difference?

Do you have any suggestion?

Thanks in advance,
Crypto

**Comment by Menchen** [comment]
My idle usage is with all program closed, and **unplug all USB device**. Right now, with brightness at max and with chrome open with 40 tabs, plus more program(airmail, discord, vscode, iterm2, and game), my usage is about 31Watts. Also I noticed in linux, without optimization nor undervolt, the battery life is similar(4h). But I'm sure that with some tweak I can achieve 6h in Linux.

**Comment by TheCrypt0** [comment]
@Menchen

I'm sorry what's your idle usage?

**Comment by Menchen** [comment]
As said in https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/116#issuecomment-511131592, about 10W, I don't remember whether I lower brightness or not.

**Comment by matvada** [comment]
I have the same problem, at idle my usage is 18.65 watts and my battery lasts about 3 hours. Any suggestion?

**Comment by Menchen** [comment]
Disable HiDPI(hight resolution need higher framebuffer and more power to render pixel.), unplug USB device, enable airplane(intel wifi still consume energy), lower brightness, undelvolt, also use h264ify if you watch youtube in chrome, disable PowerNap. Maybe enabling SpeedShift to help CPU power management, and check for background apps.

**Comment by TheCrypt0** [comment]
@Menchen 
Thanks for your suggestions. I can confirm that:
* I have HiDPI disabled.
* No USB devices attached (except for a tiny WiFi dongle that I'm sure it doesn't reach 1 watt of consumption).
* Airplane mode? Are you referring to macOS or Windows? IIRC the former doesn't have the Intel drivers to control the integrated wireless card.
* The brightness is set to 50%, reducing it even further may increase the battery life a little bit but it would become unusable on the outside. (I primarily use Windows at >50% brightness).
* My CPU, GPU and cache are currently undervolted to -115mV.
* Speedshift may make a difference. I'll enable it this evening and let you know.

Bonus points: 
* Just enabled the BIOS tweaks [DVMT_and_0xE2_fix](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/tree/master/BIOS/DVMT_and_0xE2_fix) and edited the `/EFI/CLOVER/config.plist` file but I can't see a noticeable difference.
* I noticed that the CPU's package consumption is always >2.5 watts (see the voltageshift screen in the first post), even when the CPU usage is almost 0 and the clock is 800 MHz. I'm not sure if this happens on Windows too.

**Comment by Violettx** [comment]
Yep the idle CPU package is always above 2.8 W while in Windows it's about 1.2W (with the Freq of 1.0GHz). I've checked the offset in macOS, three of them are all -0.09 V, and my idle freq is 0.8GHz. That is literally weird. I've done what I could. Maybe that's because macOS don't officially support 8550u. 
Plus, the CPU package power couldn't make a huge difference on battery life for a 1.6 W difference. I doubt there's something rather than just CPU configuration went wrong while we still cannot figure out what it is.

**Comment by buliaoyin** [comment]
Same problem to me, 16W power consumption in idle state.

I don't think it's caused by cpu configuration, must be something else...

**Comment by ZChuckMoris** [comment]
+1. 16-18W in idle. I have the same power consumption in Windows when having MSI Afterburner running in background (it prevent my GPU from sleeping). 
I guess, that _disable-external-gpu_ in clover config just doesn't disable GPU completely.

**Comment by gplast** [comment]
Hello again and thank you again for all this great effort and support.

So i also noticed that the power consumption on Mac OS is not as good as on Linux (with disabled MX150) and on Windows. I have all three operating systems install on my Notebook Pro and i like to monitor the temperature all the time on all operating systems. What i noticed is that on Windows and on Linux (with Intel only enable) the CPU was getting 36-38C on idle. The strange thing is that on Mac OS the temp was 42-44C on idle, which was the same temp that i get on Linux when i have the MX150 enable. 

I also noticed that using the Intel Power Gadget on Windows the PKG power was able to drop bellow 1 (0.90W) compared with the Mac OS where the lower power is around 2.80W.  I don't know a lot on how to patch and all these stuff that you guys know but i've been a hackintosh user for the last 7 years and i at least i'm able to test stuff, break them and fix them again :p 

So moving back to the power consumption, a couple of months ago, there was a version where the temp was good and i was able to get 4-5 hours on my device. From some point this was gone, and now i'm getting maybe 3-3,5 hours with a kind of hot laptop. So my first question is if the MX150 GPU is actually disable? Because it seems that is not.

As i was digging around to see whether there is a way to test if the GPU is off, i found this repository: https://github.com/SiddheshNan/Acer-A515-51G-Hackintosh. I noticed two file (attached) in ACPI/patched that are used to disable DGPU. I tried a lot of configuration but it seems that by copying this two files into our ACPI/patched folder the temp drops on 39C on idle and the power in the PKG is now around 2W. This is great but does that make any sense? I also have Battery Monitor for my battery and i notice that before copying the files the lowest power usage value was 12 Watts, while after copying the files i was able to get 9Watts.

I would really appreciate it if someone can test it or let me know if this is actually correct. I will test it for a couple of days of course to see also if the laptop last longer with the battery.

Sorry for the long message and thank you again. 

[SSDT-Disable_DGPU.zip](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/files/3906561/SSDT-Disable_DGPU.zip)

**Comment by buliaoyin** [comment]
@gplast Thanks for your work! The power consumption dropped from 16W to 10W after use the patch. I think there would be other unusable devices, such as intel wifi not actually disabled, maybe we can disable them like this to drop more power consumption.

**Comment by TheCrypt0** [comment]
@gplast I can confirm that by applying the SSDT patch the power usage went down by about ~3 watts. That's good stuff!

After some testing from other people I think it might be merged in the repo.

Thanks for providing this patch.

**Comment by ZChuckMoris** [comment]
@daliansky, please, check this solution with SSDT above

**Comment by clmsftsw** [comment]
Can someone explain to me how to apply this patch? 
Thank you!

**Comment by Menchen** [comment]
Download EFI directly from this repo, delete EFI/OC folder and install it.

**Comment by stevezhengshiqi** [comment]
Thanks for the work. The SSDT-Disable_DGPU actually exists in this repository for a long time since https://github.com/daliansky/XiaoMi-Pro-Hackintosh/commit/a44161a85d0c8294d288f8d5f65beddf720055db, and it was removed in https://github.com/daliansky/XiaoMi-Pro-Hackintosh/commit/5d63e340d14c77e5e0feb63ee2af9c852e65db63. Added it back to save power.
